src/engine/bufferscalers/enginebufferscalest.cpp

```diff
@@ -90,31 +93,32 @@ void EngineBufferScaleST::setScaleParameters(double base_rate,
 }
 
 void EngineBufferScaleST::onSampleRateChanged() {
-    buffer_back.clear();
+    m_bufferBack.clear();
     if (!getOutputSignal().isValid()) {
         return;
     }
     m_pSoundTouch->setSampleRate(getOutputSignal().getSampleRate());
-    const auto bufferSize = getOutputSignal().frames2samples(kSeekOffsetFrames);
-    if (bufferSize > buffer_back.size()) {
-        // grow buffer
-        buffer_back = mixxx::SampleBuffer(bufferSize);
-    }
+
     // Setting the tempo to a very low value will force SoundTouch
     // to preallocate buffers large enough to (almost certainly)
     // avoid memory reallocations during playback.
     m_pSoundTouch->setTempo(0.1);
-    m_pSoundTouch->putSamples(buffer_back.data(), kSeekOffsetFrames);
-    m_pSoundTouch->clear();
+    clear();
     m_pSoundTouch->setTempo(m_dTempoRatio);
+    clear();
```

Comment:
why the double clear? doesn't just the latter one suffice?

Reply:
Yes. removed the first one.